staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning_test.go

```diff
@@ -380,3 +382,219 @@ func TestDirectCodecEncode(t *testing.T) {
 		t.Errorf("expected group to be %v, got %v", e, a)
 	}
 }
+
+func buildUnstructuredDecodable(gvk schema.GroupVersionKind) runtime.Object {
+	obj := &unstructured.Unstructured{}
+	obj.SetGroupVersionKind(gvk)
+	return obj
+}
+
+func buildUnstructuredListDecodable(gvk schema.GroupVersionKind) runtime.Object {
+	obj := &unstructured.UnstructuredList{}
+	obj.SetGroupVersionKind(gvk)
+	return obj
+}
+
+func TestEncodeUnstructured(t *testing.T) {
+	v1GVK := schema.GroupVersionKind{
+		Group:   "crispy",
+		Version: "v1",
+		Kind:    "Noxu",
+	}
+	v2GVK := schema.GroupVersionKind{
+		Group:   "crispy",
+		Version: "v2",
+		Kind:    "Noxu",
+	}
+	elseGVK := schema.GroupVersionKind{
+		Group:   "crispy2",
+		Version: "else",
+		Kind:    "Noxu",
+	}
+	elseUnstructuredDecodable := buildUnstructuredDecodable(elseGVK)
+	elseUnstructuredDecodableList := buildUnstructuredListDecodable(elseGVK)
+	v1UnstructuredDecodable := buildUnstructuredDecodable(v1GVK)
+	v1UnstructuredDecodableList := buildUnstructuredListDecodable(v1GVK)
+	v2UnstructuredDecodable := buildUnstructuredDecodable(v2GVK)
+
+	testCases := []struct {
+		name       string
+		serializer *mockSerializer
```

Comment:
do we need this in the test struct? It's always the same value.

Reply:
removed